### PR TITLE
Make artifact analyzer obey laws of physics

### PR DESCRIPTION
**Pull request by eoineoineoin**
## About the PR

Artifact analyzer only has a non-hard fixture, so it can be dragged through walls and can get stuck inside them.

## Why / Balance

Artifact analyzer is made of fermions. Should occupy space.

## Media

Before:

https://github.com/space-wizards/space-station-14/assets/1676295/45e8dad7-8b07-4189-a027-e55f09402464

After:

https://github.com/space-wizards/space-station-14/assets/1676295/70088179-70d2-494a-bd2b-1b4c04c55278

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Artifact Analyzer now collides with walls

### CODE REVIEW

**Comment by TheShuEd** [comment]
Pls check also anomaly synchonizer

**Comment by Tayrtahn** [comment]
Won't this block being able to drag artifacts onto it?

**Comment by eoineoineoin** [comment]
The fixture filters are setup correctly so that the machine collides with walls, but artifacts can still be placed on it and detected by the console:

https://github.com/space-wizards/space-station-14/assets/1676295/7c2d26fc-e573-46f4-b6d6-dde7516945d8

**Comment by EmoGarbage404** [comment]
test fails arent erroneous

**Comment by eoineoineoin** [comment]
Oh, needed to merge master to see that failure;

It's a little unfortunate. The sprite component was `noRot=true`, and I made the fixture match, since otherwise you get mismatched physics:



I think it was a mistake to have mixed a `noRot=true` sprite with a `noRot=false` transform, but wasn't noticed because the analyzer didn't have proper physics before.

In the meantime, maps have been created where the analyzer has a non-zero rotation (this didn't make any visual difference - the machine always looked the same from any rotation), so we see asserts when loading these maps.

Is there some way the content in existing maps can be automatically migrated to reset their rotation? I could fix these maps up, but obviously don't want to create problems for forks. The alternative would be to make the sprite `noRot=false` so it matches the physics, but the sprite doesn't look great when rotated (`snapCardinals=true` didn't help)

**Comment by AJCM-git** [comment]
> Is there some way the content in existing maps can be automatically migrated to reset their rotation? I could fix these maps up, but obviously don't want to create problems for forks.

I mean, you can just bulk update them, maps are the last thing you have to worry about when thinking about conflicts cuz any update to them will conflict anyways

File: Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml [hunk 1]
```diff
@@ -39,7 +47,7 @@
         hard: False
   - type: Transform
     anchored: true
-    noRot: false
+    noRot: true
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml:50]: wouldnt this get inherited?

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Merge master and search for saved rotations just in case (ping me once you do that to merge quickly), otherwise this looks good

File: Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml [hunk 2]
```diff
@@ -25,6 +25,14 @@
...
+          !type:PhysShapeAabb
+          bounds: "-0.35,-0.35,0.35,0.35"
+        mask:
+        - Impassable
+        - HighImpassable
+        - MidImpassable
+      fix2:
         shape:
           !type:PhysShapeAabb
           bounds: "-0.35,-0.35,0.35,0.35"
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml:38]: Can't you just make the fixture hard below?

Comment (eoineoineoin) [reply Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml:38]: No, it needs two fixtures; one hard fixture for generating contacts and one non-hard fixture to detect artifacts that have been placed on the machine.

**Comment by eoineoineoin** [comment]
> Merge master and search for saved rotations just in case (ping me once you do that to merge quickly), otherwise this looks good

@AJCM-git Thanks. Merged in master again; maps I modified haven't been updated and there are no new instances of MachineArtifactAnalyzer with a saved rotation.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**